File: 10_Polymorphism.py
# Polymorphism allows objects of different classes to be treated as objects of a common super class. It can also refer to the ability of a function or method to work on objects of different types.


class Animal :
    def sound(self):
        print('Animal makes a sound.')
        
class Dog (Animal):
    def sound(self):
        print('Dog barks.')

an = Animal()
dog = Dog()

for ani in (an, dog):
    ani.sound()

File: 10_Polymorphism.cpp


#include <bits/stdc++.h>
using namespace std;

class Animal{
    public:
    virtual void sound(){
        cout<<"Animal sound started"<<endl;
    }

};

class Dog : public Animal{
    public:
    void sound(){
        cout<<"Dog barks"<<endl;
    }
};


int main()
{

    Animal* animal = new Animal;
    Dog* dog = new Dog;


    // animal->sound();
    // dog->sound();

    Dog dog2;
    Animal * an2;

    an2 = &dog2;
    an2->sound();
    
    return 0;
}